### ISSUE TITLE
identify().get() not complete

### ISSUE THREAD

**Issue by LukasVykuka** [launchdarkly/android-client-sdk#86]
**Describe the bug**
Call to `identify().get()` never return if it's called from another thread than main thread and app is on backgorund at the same time.

It should be possible to identify user without blocking main thread even if app is on background.

**To reproduce**
Make Android project and put this activity in it. Then create layout with `run_button`.
```
class MainActivity : AppCompatActivity(R.layout.activity_main) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val ldConfig = LDConfig.Builder()
            .setMobileKey(API_KEY)
            .build()

        val ldUser = LDUser.Builder("1234").build()
        val ld = LDClient.init(application, ldConfig, ldUser).get()

        run_button.setOnClickListener {
            thread(start = true) {
                Log.d("LD", "delay")
                TimeUnit.SECONDS.sleep(3)

                Log.d("LD", "identify")
                ld.identify(LDUser.Builder("1234").build()).get()

                Log.d("LD", "completed")
            }
        }
    }
}
```

After cklick on run button you have 3 second to put app to background. Then observe logcat and you will see `identify` log but not `completed` log. If you click on button and don't put app to background then you will see both logs.

**Expected behavior**
Return from `identify().get()` method even if called from background thread if app is not in foreground.

**SDK version**
2.8.5

**Language version, developer tools**
Kotlin 1.3.50

**OS/platform**
Android

**Comment by gwhelanLD** [comment]
Hi @LukasVykuka,

Thanks for reporting this issue. We'll investigate and should have a fix out with the next release.

Thanks,
@gwhelanLD

**Comment by LukasVykuka** [comment]
I have another information - if blocking `TimeUnit.SECONDS.sleep(3)` is replaced by non-blocking `Handler().postDelayed()` than `identify().get()` never return even on main thread if app is on background. Here is code snippet to test it. To reproduce follow steps mentioned in original post.
```
run_button.setOnClickListener {
            Log.d("LD", "delay")
            Handler().postDelayed(
                {
                    Log.d("LD", "identify")
                    ld.identify(LDUser.Builder("1234").build()).get()

                    Log.d("LD", "completed")
                },
                3000
            )
        }
```

**Comment by LukasVykuka** [comment]
Is seems same as this issue https://github.com/launchdarkly/android-client-sdk/issues/73

**Comment by LukasVykuka** [comment]
Is there any progress?

**Comment by gwhelanLD** [comment]
Firstly, thank you for your patience so far on this issue. We've identified the issue, and currently have a fix in testing to be released soon.

Thanks,
@gwhelanLD

**Comment by LukasVykuka** [comment]
That's great news, thank you!

**Comment by gwhelanLD** [comment]
Thanks again for your patience on this issue @LukasVykuka. I wanted to follow up that we just released version 2.9.0 with a fix for the issue with the Future completion timing when the application is backgrounded. Don't hesitate to reach out if you run into any other issue.

Thanks,
-- @gwhelanLD

**Comment by LukasVykuka** [comment]
I can confirm that issue is resolved in 2.9.0. Thank you for it.

**Comment by drommk** [comment]
Are you sure it's correctly fixed? I'm facing a similar issue with 2.9.0. 

I basically have wrapped all LD interactions in Rx streams. And at some point it involves having `identify` calls running on a specific (non-main) thread. 

But if the app goes into the background when `identify()` is being called, the call is [cancelled](https://github.com/launchdarkly/okhttp-eventsource/blob/master/src/main/java/com/launchdarkly/eventsource/EventSource.java#L191), the event, the connection is [shut down](https://github.com/launchdarkly/okhttp-eventsource/blob/master/src/main/java/com/launchdarkly/eventsource/EventSource.java#L278)

And `identify().get()` will never return, no matter if I retry it after a timeout.

I had to downgrade to `2.7.0` to have it work as expected.

**Comment by gwhelanLD** [comment]
@drommk We believe this is a separate issue, and filed #91 to track it.
